homeassistant/components/evohome/water_heater.py

```diff
@@ -87,15 +98,19 @@ def __init__(self, evo_broker, evo_device) -> None:
         )
 
     @property
-    def current_operation(self) -> str:
+    def current_operation(self) -> str | None:
         """Return the current operating mode (Auto, On, or Off)."""
         if self._evo_device.mode == EVO_FOLLOW:
             return STATE_AUTO
+        if self._evo_device.state is None:
```

Comment:
We could store the value in a local variable via a walrus to avoid another attribute access below.

Reply:
Well, we _could_, but I don't think this is a case where it adds value:
 - it doesn't save lines of code, nor remove indentation 
 - it doesn't mean that a significant amount of processing is only done once
 
Also, I don't think that:
```python
EVO_STATE_TO_HA[self._evo_device.state]
```
... offers any value over:
```python
EVO_STATE_TO_HA[device_state]
```

Reply:
I think it's part of general code discipline. If we follow that we don't need to worry about doing more work than needed when it would matter.

Reply:
Hmmm... Honestly: I am not sure if I agree. But PEP 572 has been like that for many, I guess, so it's something I'll reflect upon.

I'm happy to change it as you request.